File: .NET/VS2010TrainingKit/Labs/OfficeUICustomization/Source/Ex2-WPFTaskPane/Begin/VB/Snippet.vb
' ----------------------------------------------------------------------------------
' Microsoft Developer & Platform Evangelism
' 
' Copyright (c) Microsoft Corporation. All rights reserved.
' 
' THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
' OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
' ----------------------------------------------------------------------------------
' The example companies, organizations, products, domain names,
' e-mail addresses, logos, people, places, and events depicted
' herein are fictitious.  No association with any real company,
' organization, product, domain name, email address, logo, person,
' places, or events is intended or should be inferred.
' ----------------------------------------------------------------------------------

Public Class Snippet
    Sub New()
    End Sub

    Sub New(ByVal name As String, ByVal description As String, ByVal content As String)
        Me.Name = name
        Me.Description = description
        Me.Content = content
    End Sub

    Public Property Name As String
    Public Property Description As String
    Public Property Content As String
End Class

File: .NET/VS2010TrainingKit/Labs/OfficeUICustomization/Source/Ex2-WPFTaskPane/Begin/C#/Snippet.cs
// ----------------------------------------------------------------------------------
// Microsoft Developer & Platform Evangelism
// 
// Copyright (c) Microsoft Corporation. All rights reserved.
// 
// THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
// OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
// ----------------------------------------------------------------------------------
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
// ----------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Windows.Threading;

namespace WPFTaskPane
{   
    public class Snippet
    {
        public Snippet()
        { }

        public Snippet(string name, string description, string content)
        {
            Name = name;
            Description = description;
            Content = content;
        }

        public string Name { get; set; }
        public string Description { get; set; }
        public string Content { get; set; }
    }
}